tardis/io/tests/test_config_reader.py

```diff
@@ -13,6 +14,48 @@ def data_path(filename):
     data_dir = os.path.dirname(__file__)
     return os.path.abspath(os.path.join(data_dir, 'data', filename))
 
+def traverse_configs(base, other, func, *args):
+    """
+    Recursively traverse a base dict or list along with another one
+    calling `func` for leafs of both objects.
+
+    Parameters
+    ----------
+    base:
+        The object on which the traversing is done
+    other:
+        The object which is traversed along with `base`
+    func:
+        A function called for each leaf of `base` and the correspnding leaf of `other`
+        Signature: `func(item1, item2, *args)`
+    args:
+        Arguments passed into `func`
+
+    """
+    if isinstance(base, collections.Mapping):
+        for k in base:
+            traverse_configs(base[k], other[k], func, *args)
+    elif isinstance(base, collections.Iterable) and not isinstance(base, basestring) and not hasattr(base, 'shape'):
+        for val1, val2 in zip(base, other):
+            traverse_configs(val1, val2, func, *args)
+    else:
+        func(base, other, *args)
+
+
+def check_equality(item1, item2):
```

Comment:
The implementation looks good but currently `traverse_configs(default_conf, custom_conf, check_equality)` would not work because you are not returning anything in `traverse_configs`. That's why I suggested using `assert_equality` as your function that gets passed to `traverse_configs`. Then you could implement `check_equality` as a simple `try: ...; except AssertionError: return False; else: return True;`. In it's current implementation I don't see `check_equality` working.